### ISSUE TITLE
No chromosome_density_plots.pdf

### ISSUE THREAD

**Issue by bijendrabio** [oushujun/EDTA#514]
Hello @oushujun @FayeFang17 ,
Although the EDTA run completed successfully, I received the following error message and couldn’t find chromosome_density_plots.pdf. Is it possible to generate chromosome_density_plots.pdf from the output?

Errror
Fri Oct  4 18:50:43 EDT 2024	Execution of EDTA_raw.pl is finished!
Sun Oct  6 02:04:23 EDT 2024	Homology-based annotation of TEs using testgenome.EDTA.fasta.mod.EDTA.TElib.fa from scratch.

mv: cannot stat 'chromosome_density_plots.pdf': No such file or directory
Use of uninitialized value $mod_time in localtime at /tools/EDTA/EDTA.pl line 847.
cp: cannot stat ' testgenome.fasta.mod.EDTA.TEanno.density_plots.pdf': No such file or directory

Regards,
B

**Comment by Jdbeck66** [comment]
Similar. 

Error encountered: [Errno 2] No such file or directory: 'samtools'
mv: cannot stat 'chromosome_density_plots.pdf': No such file or directory
Sat Nov  2 09:20:53 MDT 2024	TE annotation using the EDTA library has finished! Check out:
		Whole-genome TE annotation (total TE: 64.47%): BEA_div10.purged.fa.mod.EDTA.TEanno.gff3
		Whole-genome TE annotation summary: BEA_div10.purged.fa.mod.EDTA.TEanno.sum
		Whole-genome TE divergence plot: BEA_div10.purged.fa.mod_divergence_plot.pdf
		Whole-genome TE density plot: BEA_div10.purged.fa.mod.EDTA.TEanno.density_plots.pdf
		Low-threshold TE masking for MAKER gene annotation (masked: 39.69%): BEA_div10.purged.fa.mod.MAKER.masked

Use of uninitialized value $mod_time in localtime at /lfs/jimbeck.bsu/EDTA/EDTA.pl line 847.
cp: cannot stat 'BEA_div10.purged.fa.mod.EDTA.TEanno.density_plots.pdf': No such file or directory

Used the build suggested in issue #483 - I'm thinking `samtools` should be included in:

`mamba create -n EDTA2.2 -c conda-forge -c bioconda -c r annosine2 biopython blast cd-hit coreutils genericrepeatfinder genometools-genometools glob2 h5py==3.9 keras==2.11 ltr_finder ltr_retriever mdust multiprocess muscle openjdk pandas perl perl-text-soundex pyarrow python r-base r-dplyr regex repeatmodeler r-ggplot2 r-here r-tidyr scikit-learn swifter tensorflow==2.11 tesorter`

**Comment by WWilen** [comment]
Hello @oushujun @FayeFang17 ，I have installed all the software as prompted #483 ，but
the sam problem still exists.

2024年 11月 16日 星期六 00:03:47 CST    Execution of EDTA_raw.pl is finished!


Warning: No sequences were masked
2024年 11月 16日 星期六 00:04:07 CST    Homology-based annotation of TEs using genome.fa.mod.EDTA.TElib.fa from scratch.

mv: 对 'chromosome_density_plots.pdf' 调用 stat 失败: 没有那个文件或目录
Use of uninitialized value $mod_time in localtime at ../EDTA.pl line 847.
cp: 对 'genome.fa.mod.EDTA.TEanno.density_plots.pdf' 调用 stat 失败: 没有那个文件或目录

**Comment by WWilen** [comment]
In addition,I installed it by downloading zip

**Comment by WWilen** [comment]
I also tried different servers and different EDTA2 versions，the problem still exists

**Comment by duanyajuan** [comment]
I'm having the same problem.
#########################################################
##### Extensive de-novo TE Annotator (EDTA) v2.2.1  #####
##### Shujun Ou (shujun.ou.1@gmail.com)             #####
#########################################################


Parameters: --genome genome.fa --cds genome.cds.fa --curatedlib ../database/rice7.0.0.liban --exclude genome.exclude.bed --overwrite 1 --sensitive 1 --anno 1 --threads 10


2024年 11月 21日 星期四 23:03:18 CST	Dependency checking:
	All passed!

	A custom library ../database/rice7.0.0.liban is provided via --curatedlib. Please make sure this is a manually curated library but not machine generated.

	A CDS file genome.cds.fa is provided via --cds. Please make sure this is the DNA sequence of coding regions only.

	A BED file is provided via --exclude. Regions specified by this file will be excluded from TE annotation and masking.

2024年 11月 21日 星期四 23:03:20 CST	Obtain raw TE libraries using various structure-based programs: 
2024年 11月 21日 星期四 23:03:20 CST	EDTA_raw: Check dependencies, prepare working directories.

2024年 11月 21日 星期四 23:03:21 CST	Start to find LTR candidates.

2024年 11月 21日 星期四 23:03:21 CST	Identify LTR retrotransposon candidates from scratch.

Warning: LOC list genome.fa.mod.ltrTE.veryfalse is empty.
2024年 11月 21日 星期四 23:03:48 CST	Finish finding LTR candidates.

2024年 11月 21日 星期四 23:03:48 CST	Start to find SINE candidates.

2024年 11月 21日 星期四 23:04:43 CST	Warning: The SINE result file has 0 bp!

2024年 11月 21日 星期四 23:04:43 CST	Start to find LINE candidates.

2024年 11月 21日 星期四 23:04:43 CST	Identify LINE retrotransposon candidates from scratch.

2024年 11月 21日 星期四 23:06:14 CST	Warning: The LINE result file has 0 bp!

2024年 11月 21日 星期四 23:06:14 CST	Start to find TIR candidates.

2024年 11月 21日 星期四 23:06:14 CST	Identify TIR candidates from scratch.

Species: others
2024年 11月 21日 星期四 23:06:43 CST	Finish finding TIR candidates.

2024年 11月 21日 星期四 23:06:43 CST	Start to find Helitron candidates.

2024年 11月 21日 星期四 23:06:43 CST	Identify Helitron candidates from scratch.

2024年 11月 21日 星期四 23:07:16 CST	Finish finding Helitron candidates.

2024年 11月 21日 星期四 23:07:16 CST	Execution of EDTA_raw.pl is finished!

2024年 11月 21日 星期四 23:07:16 CST	Obtain raw TE libraries finished.
	All intact TEs found by EDTA: 
		genome.fa.mod.EDTA.intact.raw.fa 
		genome.fa.mod.EDTA.intact.raw.gff3

2024年 11月 21日 星期四 23:07:16 CST	Perform EDTA advance filtering for raw TE candidates and generate the stage 1 library: 


Warning: No sequences were masked
2024年 11月 21日 星期四 23:07:29 CST	EDTA advance filtering finished.

2024年 11月 21日 星期四 23:07:29 CST	Perform EDTA final steps to generate a non-redundant comprehensive TE library.

	Filter RepeatModeler results that are ignored in the raw step.

2024年 11月 21日 星期四 23:07:34 CST	Clean up TE-related sequences in the CDS file with TEsorter.

	Remove CDS-related sequences in the EDTA library.

	Remove CDS-related sequences in intact TEs.

2024年 11月 21日 星期四 23:07:44 CST	Combine the high-quality TE library rice7.0.0.liban with the EDTA library:

2024年 11月 21日 星期四 23:07:54 CST	EDTA final stage finished! You may check out:
		The final EDTA TE library: genome.fa.mod.EDTA.TElib.fa
		Family names of intact TEs have been updated by rice7.0.0.liban: genome.fa.mod.EDTA.intact.gff3
	Comparing to the provided library, EDTA found these novel TEs: genome.fa.mod.EDTA.TElib.novel.fa
		The provided library has been incorporated into the final library: genome.fa.mod.EDTA.TElib.fa

2024年 11月 21日 星期四 23:07:54 CST	Perform post-EDTA analysis for whole-genome annotation:

2024年 11月 21日 星期四 23:07:54 CST	Homology-based annotation of TEs using genome.fa.mod.EDTA.TElib.fa from scratch.

mv: 对 'chromosome_density_plots.pdf' 调用 stat 失败: 没有那个文件或目录
2024年 11月 21日 星期四 23:08:04 CST	TE annotation using the EDTA library has finished! Check out:
		Whole-genome TE annotation (total TE: 34.61%): genome.fa.mod.EDTA.TEanno.gff3
		Whole-genome TE annotation summary: genome.fa.mod.EDTA.TEanno.sum
		Whole-genome TE divergence plot: genome.fa.mod_divergence_plot.pdf
		Whole-genome TE density plot: genome.fa.mod.EDTA.TEanno.density_plots.pdf
		Low-threshold TE masking for MAKER gene annotation (masked: 17.27%): genome.fa.mod.MAKER.masked

Use of uninitialized value $mod_time in localtime at ../EDTA.pl line 847.
cp: 对 'genome.fa.mod.EDTA.TEanno.density_plots.pdf' 调用 stat 失败: 没有那个文件或目录
2024年 11月 21日 星期四 23:08:04 CST	Evaluate the level of inconsistency for whole-genome TE annotation:

2024年 11月 21日 星期四 23:08:07 CST	Evaluation of TE annotation finished! Check out these files:

				Overall: genome.fa.mod.EDTA.TE.fa.stat.all.sum
				Nested: genome.fa.mod.EDTA.TE.fa.stat.nested.sum
				Non-nested: genome.fa.mod.EDTA.TE.fa.stat.redun.sum

		If you want to learn more about the formatting and information of these files, please visit:
			https://github.com/oushujun/EDTA/wiki/Making-sense-of-EDTA-usage-and-outputs---Q&A

**Comment by oushujun** [comment]
Hello, please install the latest update made today, both the conda env and the code. It should fix your issue. Please let me know otherwise.

Thanks!
Shujun
